### PR TITLE
Handle attachments from web sockets

### PR DESCRIPTION
**Pull request by AndriiHorishniiMOC**
Replace an attachment for the visitor messages with an attachment from messages received from web sockets if a message was sent using the secure conversation endpoint.

[MOB-2046](https://glia.atlassian.net/browse/MOB-2046)

[MOB-2046]: https://glia.atlassian.net/browse/MOB-2046?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Pelkar — APPROVED** [review]
LGTM, the "Unengagement" seems like a weird term, but I see the need for it and I could not  think of anything better

File: widgetssdk/src/main/java/com/glia/widgets/chat/domain/UnengagementMessageUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import com.glia.androidsdk.chat.ChatMessage
+import com.glia.widgets.chat.data.GliaChatRepository
+import com.glia.widgets.core.engagement.GliaEngagementRepository
+import com.glia.widgets.core.engagement.domain.GliaOnEngagementUseCase
+import com.glia.widgets.core.engagement.domain.MapOperatorUseCase
+import com.glia.widgets.core.engagement.domain.model.ChatMessageInternal
+import io.reactivex.Observable
+import java.util.function.Consumer
+
+class UnengagementMessageUseCase(
```
Comment (gugalo) [inline widgetssdk/src/main/java/com/glia/widgets/chat/domain/UnengagementMessageUseCase.kt:12]: I like your name for repository: allMessage, maybe we could name it a similar way, e.g: `AllMessageUseCase` or maybe even `GlobalMessageUseCase`

Comment (gugalo) [reply widgetssdk/src/main/java/com/glia/widgets/chat/domain/UnengagementMessageUseCase.kt:12]: Same in other places

Comment (AndriiHorishniiMOC) [reply widgetssdk/src/main/java/com/glia/widgets/chat/domain/UnengagementMessageUseCase.kt:12]: It unsubscribes from events after an engagement starts. So, I don't find the `AllMessageUseCase` as a good name for this use case.

**Review by DavDo — APPROVED** [review]
LGTM

**Comment by AndriiHorishniiMOC** [comment]
!merge

**Pull request merged**